### PR TITLE
add lodash types

### PR DESCRIPTION
**Pull request by wrexbe**
- Added lodash types
- Deadly Dirge won't waste attacks on dead people
- Fix steal helper (maybe? I didn't test it, but I can't see how the previous code would of worked either)
- Fix bug in Dispel helper

### CODE REVIEW

File: server/src/helpers/character/CharacterHelper.ts [hunk 1]
```diff
@@ -541,7 +541,7 @@ export class CharacterHelper extends BaseService {
     }
 
     // can't move more than one screen at a time
-    character.totalStats[Stat.Move] = clamp(character.totalStats[Stat.Move], 0, 4);
+    character.totalStats[Stat.Move] = clamp(character.totalStats.move ?? 0, 0, 4);
```
Comment (seiyria) [inline server/src/helpers/character/CharacterHelper.ts:544]: why was this changed? Stat.Move is valid

File: server/src/models/world/Spawner.ts [hunk 2]
```diff
@@ -361,7 +361,7 @@ export class Spawner {
 
   public getRandomPath(): string {
     if (!this.hasPaths) return '';
-    return sample(this.paths);
+    return sample(this.paths) ?? '';
```
Comment (seiyria) [inline server/src/models/world/Spawner.ts:364]: this function could return string | undefined I believe instead

File: server/src/helpers/character/StealHelper.ts [hunk 3]
```diff
@@ -92,12 +92,13 @@ export class StealHelper extends BaseService {
       const handName = this.game.characterHelper.getEmptyHand(char);
       if (!handName) return;
 
-      const item = sample(target.items.sack.items);
-      this.game.inventoryHelper.removeItemFromSack(target, item);
+      const item = sample(target.items.sack.items) as ISimpleItem;
```
Comment (seiyria) [inline server/src/helpers/character/StealHelper.ts:95]: can you add an if(!item) return this.sendMessage('you could not find anything to take!')

**Review by seiyria — CHANGES_REQUESTED** [review]

File: server/src/helpers/character/StealHelper.ts (hunk 3 above)
Comment (wrexbe) [reply server/src/helpers/character/StealHelper.ts:95]: It does this above that
else if (target.items.sack.items.length > 0)

File: server/src/models/world/Spawner.ts (hunk 2 above)
Comment (wrexbe) [reply server/src/models/world/Spawner.ts:364]: Nope, that didn't work

File: server/src/helpers/character/CharacterHelper.ts (hunk 1 above)
Comment (wrexbe) [reply server/src/helpers/character/CharacterHelper.ts]: Ok, changed back

File: server/src/models/world/Spawner.ts (hunk 2 above)
Comment (wrexbe) [reply server/src/models/world/Spawner.ts:364]: Alternative didn't work

File: server/src/helpers/character/StealHelper.ts (hunk 3 above)
Comment (wrexbe) [reply server/src/helpers/character/StealHelper.ts:95]: It does a check at the very top. That being said this method has other issues that should be fixed in a new PR.
